x-pack/plugins/alerting/common/rule.ts

```diff
@@ -76,6 +76,20 @@ export interface RuleExecutionStatus {
 export type RuleActionParams = SavedObjectAttributes;
 export type RuleActionParam = SavedObjectAttribute;
 
+export interface AlertsFilter extends SavedObjectAttributes {
+  query: null | {
+    kql: string;
+    dsl?: string;
```

Comment:
Added "dsl" as optional o prevent 10s of files from being modified.
Actually it's not optional, create, update and bulkEdit methods of the rulesClient always generates this filed out of kql and saves in ES.

To avoid this hack, we need to refactor our RuleAction interfaces.

Reply:
Can we create an issue for this refactor so we can document which fields are currently typed as optional but not actually optional? Might be good to leave a code comment for the `dsl` field as well

Reply:
Created an issue for that: https://github.com/elastic/kibana/issues/153535